include/kernel.h

```diff
@@ -776,6 +780,8 @@ extern FUNC_NORETURN void k_thread_user_mode_enter(k_thread_entry_t entry,
  * need to be.
  * Note that NULL shouldn't be passed as an argument.
  *
+ * @allowed_from Supervisor Mode
```

Comment:
Everything is allowed from supervisor mode.
I think it's more useful to indicate which core kernel APIs are callable from user mode (although I suspect nearly all or all of them are tagged with __syscall)